client/src/utils/request.js

```diff
@@ -1,3 +1,27 @@
-export default function request(url) {
-    return fetch(url).then(response => response.json());
+export default function request(url, method, data) {
```

Comment:
It would be much easier to use if your module exposed separate `get` and `post` methods.